test/Concurrency/Runtime/async_taskgroup_dontLeakTasks.swift

```diff
@@ -0,0 +1,43 @@
+// RUN: %target-run-simple-swift( -Xfrontend -disable-availability-checking -parse-as-library) | %FileCheck %s
+// REQUIRES: executable_test
+// REQUIRES: concurrency
+// UNSUPPORTED: use_os_stdlib
+// UNSUPPORTED: back_deployment_runtime
+// UNSUPPORTED: linux
+
+#if os(Linux)
+import Glibc
+#elseif os(Windows)
+import MSVCRT
+#else
+import Darwin
+#endif
+
+@available(SwiftStdlib 5.5, *)
+func test_taskGroup_next() async {
+  _ = await withTaskGroup(of: Int.self, returning: Int.self) { group in
+    for n in 0..<100 {
+      group.spawn {
+        return n
+      }
+    }
+    await Task.sleep(2_000_000)
+
+    var sum = 0
+    for await value in group {
+      sum += 1
+    }
+
+    return sum
+  }
+
+  // CHECK: result with group.next(): 100
+  print("result with group.next(): \(100)")
+}
+
+@available(SwiftStdlib 5.5, *)
+@main struct Main {
+  static func main() async {
+    await test_taskGroup_next()
+  }
```

Comment:
no great way to test this without having task `destroy` print... is there some trick we could pull here?

Reply:
You could put the tasks into an array of weak references, then check that they're all nil after you're done?

Reply:
Oh, let me try that!

Reply:
Hmm... it's pretty tricky, Task isn't really just a class, but we'd have to use the Builtin.NativeObject which we can't here... but trying to pretend it is AnyObject gets me into compiler crashes 😢  

There is no way to just "get a task" as a class; there is only `withUnsafeCurrentTask` that gives us a struct wrapper around the native object. I'm not sure it'd even work (I'll try a little bit more though)